src/main/java/net/dv8tion/jda/internal/entities/TextChannelImpl.java

```diff
@@ -458,14 +458,23 @@ public MessageAction sendFile(@Nonnull InputStream data, @Nonnull String fileNam
         Checks.isSnowflake(messageId, "Message ID");
         Checks.noWhitespace(unicode, "Unicode emoji");
         Checks.notNull(user, "User");
+
         if (!getJDA().getSelfUser().equals(user))
             checkPermission(Permission.MESSAGE_MANAGE);
-        final String code = EncodingUtil.encodeUTF8(unicode);
-        Route.CompiledRoute route;
+
+        String encoded;
+        if (unicode.startsWith("U+"))
```

Comment:
It's probably better to also check for lowercase version.

Reply:
Good catch, will replace it with `unicode.toLowerCase().startsWith("u+")` so no second if is needed.

Reply:
It would be more efficient to just use a logical or than to lowercase the entire string

Reply:
Gotcha. Just tried to avoid another check since I thought the string would be at most like 6 chars so lowering it wouldn't matter too much.